### PR TITLE
When using Secrets, config functionality should be auto enabled

### PR DESCRIPTION
**Pull request by ketoketo**
* is related to #11258

### CODE REVIEW

**Comment by geoand** [comment]
Thanks for this!

Can you please check the output of CI? Something is failing

**Comment by geoand** [comment]
The relevant CI failure is:

```
2020-08-31T14:40:00.2255552Z [INFO] Running io.quarkus.it.kubernetes.client.SecretPropertiesTest
2020-08-31T14:40:05.7507108Z 2020-08-31 14:40:02,825 WARN  [io.qua.dep.QuarkusAugmentor] (main) Using Java versions older than 11 to build Quarkus applications is deprecated and will be disallowed in a future release!
2020-08-31T14:40:05.7507902Z 2020-08-31 14:40:05,156 INFO  [okh.moc.MockWebServer] (MockWebServer 55545) MockWebServer[55545] starting to accept connections
2020-08-31T14:40:05.7508464Z 2020-08-31 14:40:05,160 WARN  [io.fab.kub.cli.Config] (main) Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.
2020-08-31T14:40:05.7509100Z 2020-08-31 14:40:05,499 WARN  [io.qua.kub.cli.run.KubernetesConfigRecorder] (main) Configuration is read from Secrets [s1], but quarkus.kubernetes-config.secrets.enabled is false. Check if your application's service account has enough permissions to read secrets.
2020-08-31T14:40:05.7509836Z 2020-08-31 14:40:05,560 INFO  [okh.moc.MockWebServer] (MockWebServer /127.0.0.1:58024) MockWebServer[55545] received request: GET /api/v1/namespaces/test/configmaps/cmap1 HTTP/1.1 and responded: HTTP/1.1 200 OK
2020-08-31T14:40:05.7510801Z 2020-08-31 14:40:05,627 INFO  [okh.moc.MockWebServer] (MockWebServer /127.0.0.1:58024) MockWebServer[55545] received request: GET /api/v1/namespaces/test/configmaps/cmap2 HTTP/1.1 and responded: HTTP/1.1 200 OK
2020-08-31T14:40:05.7511384Z 2020-08-31 14:40:05,725 ERROR [io.qua.application] (main) Failed to start application (with profile test): javax.enterprise.inject.spi.DeploymentException: No config value of type [java.lang.String] exists for: secret.prop1
2020-08-31T14:40:05.7511642Z 	at io.quarkus.arc.runtime.ConfigRecorder.validateConfigProperties(ConfigRecorder.java:37)
```

**Comment by ketoketo** [comment]
Thanks for the review. I'll try to fix it.

**Comment by geoand** [comment]
Can you please squash the commits?

**Comment by geoand** [comment]
There is one thing I don't like: With the current approach, having `quarkus.kubernetes-config.enabled=true` also enables configmap reading, which is something we probably don't want

**Comment by ketoketo** [comment]
I thought that being able to read configMap when `quarkus.kubernetes-config.enabled=true` was the same behavior as before, but am I mistaken?

**Comment by geoand** [comment]
Sorry, I meant that when enabling secrets, then config maps are enabled - that is what I am saying is a little weird

File: extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceProvider.java [hunk 1]
```diff
@@ -38,10 +41,10 @@ public KubernetesConfigSourceProvider(KubernetesConfigSourceConfig config, Kuber
         }
 
         List<ConfigSource> result = new ArrayList<>();
-        if (config.configMaps.isPresent()) {
+        if (config.enabled && config.configMaps.isPresent()) {
```
Comment (ketoketo) [inline extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceProvider.java:44]: >enabling secrets, then config maps are enabled

I believe I've modified the configMap to not load the configMap if `quarkus.kubernetes-config.enabled=false`. What do you think?

Comment (geoand) [reply extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceProvider.java:44]: I'll take a closer look tomorrow

Comment (geoand) [reply extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceProvider.java:44]: Ah right, I hadn't seen this!

File: extensions/kubernetes-config/runtime/src/test/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceProviderTest.java [hunk 2]
```diff
@@ -13,10 +14,7 @@
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
-import io.fabric8.kubernetes.api.model.ConfigMap;
-import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
-import io.fabric8.kubernetes.api.model.ConfigMapList;
-import io.fabric8.kubernetes.api.model.DoneableConfigMap;
+import io.fabric8.kubernetes.api.model.*;
```
Comment (geoand) [inline extensions/kubernetes-config/runtime/src/test/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceProviderTest.java:17]: Minor nitpick: Please avoid using start imports

Comment (ketoketo) [reply extensions/kubernetes-config/runtime/src/test/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceProviderTest.java]: Fixed it

**Review by geoand — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**